pycaching/geocaching.py

```diff
@@ -356,6 +364,49 @@ def search_quick(self, area, *, strict=False, zoom=None):
 
     # add some shortcuts ------------------------------------------------------
 
+    def search_rect(self, rect: Rectangle, *, per_query: int = 50, sort_by: str = 'datelastvisited', origin=None):
+        """
+        Return a generator of caches in given Rectange area.
+
+        :param rect: Search area.
+        :param per_query: Number of caches requested in single query.
+        :param sort_by: Order cached by given criterion.
+        :param origin: Origin point for search by distance.
+        """
+        if not isinstance(sort_by, SortOrder):
+            sort_by = SortOrder(sort_by)
+
+        params = {
+            'box': "{},{},{},{}".format(
+                rect.corners[0].latitude, rect.corners[0].longitude,
+                rect.corners[1].latitude, rect.corners[1].longitude),
+            'take': per_query,
+            'asc': 'true',
+            'skip': 0,
+            'sort': sort_by.value,
+        }
+
+        if sort_by == SortOrder.distance:
+            assert isinstance(origin, Point)
+            params['origin'] = '{},{}'.format(origin.latitude, origin.longitude)
+
+        total, offset = None, 0
+        while (total is None) or (offset < total):
+            params['skip'] = offset
+
+            try:
+                resp = self._request(self._urls['api_search'], params=params, expect='json')
+            except TooManyRequests as e:
+                import time
+                time.sleep(e.rate_limit_reset + 5)
```

Comment:
This is neat for most cases.

But as a pycaching user, I would like to have a control over rate-limit behavior. For example, I may want to do something else instead of sleeping and get back to downloading caches later.

There should be some parameter allowing propagation of this exception and some way of resuming the search loop. From implementation point of view, I have no preference how to solve it.

Reply:
:+1:  done. I have introduced `wait_sleep` flag, which if set to `False` will result in yielding `None`s instead of sleeping. Resuming generator broken with `Exceptions` seems to be not possible.

Reply:
The parameter name is not very descriptive. :) But I like the solution from technical point of view. 👍